app/main/views/questionnaire.py

```diff
@@ -115,26 +125,37 @@ def redirect_to_questionnaire_page(eq_id, form_type, collection_id, location):
                             location=location))
 
 
-def same_survey(eq_id, form_type, collection_id):
+def _same_survey(eq_id, form_type, collection_id):
     metadata = get_metadata(current_user)
     current_survey = eq_id + form_type + collection_id
     metadata_survey = metadata["eq_id"] + metadata["form_type"] + metadata["collection_exercise_sid"]
     return current_survey == metadata_survey
 
 
-def render_page(location, is_valid):
-    context = g.questionnaire_manager.get_rendering_context(location, is_valid)
+def _render_page(location, context):
     template = TemplateRegistry.get_template_name(location)
     navigator = g.questionnaire_manager.navigator
     previous_location = navigator.get_previous_location(get_answers(current_user), location)
-    return render_template(template, context, previous_location)
+    return _render_template(template, context, previous_location)
```

Comment:
I think we need to rationalise these methods `_render_page > render_template > render_theme_template`, each one is returning a call to another method. How about we refactor `_render_template` into a method to get the theme, and call `render_theme_template` from this method?